src/cookbooks/clients_last_seen_bits.md

```diff
@@ -0,0 +1,136 @@
+# Working with Bit Patterns in Clients Last Seen
+
+An powerful feature of the `clients_last_seen` methodology is that we can capture
+per-day usage information for the entire 28-day period that each row in the table
+describes. We encode usage information in "bit patterns" where the physical
+type of the field is a BigQuery 64-bit integer, but logically the integer
+represents an array of bits, with each 1 indicating a day where the given clients
+was active and each 0 indicating a day where the client was inactive.
+
+The representation is space-efficient and actually easier to manipulate
+in BigQuery than a date array would be, but requires some practice to interact
+with directly. For well-established usage and retention metrics, we generally
+provide logic in views to insulate users from having to ever consider the
+raw bit patterns, but new investigations may require novel retention definitions.
+
+## Usage: Backward-looking windows
```

Comment:
Don't we look forward in GUD?  For the record I prefer backward.  It raises less questions from folks that wonder where the recent data is.

Reply:
WAU and MAU are backward-looking metrics where the value for 2020-01-28 depends on activity from 2020-01-01 to 2020-01-28.

Retention metrics are forward-looking where the value for 2020-01-28 depends on activity happening _after_ that date.

This is confusing and subtle for those who have to calculate the metrics, but I do think it's the most straight-forward approach from the consumer side. For a given date, you may want to compare 3-day retention, 1-week retention, and 2-week retention; this only makes sense if we align them based on the first day of the window.

Reply:
I added a new "Forward-looking windows and backward-looking windows" section above this to hopefully capture the gotchas here.